node/subsystem/src/lib.rs

```diff
@@ -228,3 +228,33 @@ impl<C: SubsystemContext> Subsystem<C> for DummySubsystem {
 		}
 	}
 }
+
+/// A forward subsystem that implements [`Subsystem`].
+///
+/// It forwards all communication from the overseer to the internal message
+/// channel.
+///
+/// This susbsytem is useful for testing functionality that interacts with the overseer.
+pub struct ForwardSubsystem<Msg>(pub mpsc::Sender<Msg>);
+
+impl<C: SubsystemContext<Message = Msg>, Msg: Send + 'static> Subsystem<C> for ForwardSubsystem<Msg> {
+	fn start(mut self, mut ctx: C) -> SpawnedSubsystem {
+		let future = Box::pin(async move {
+			loop {
+				match ctx.recv().await {
+					Ok(FromOverseer::Signal(OverseerSignal::Conclude)) => return,
+					Ok(FromOverseer::Communication { msg }) => {
+						let _ = self.0.send(msg).await;
+					},
+					Err(_) => return,
+					_ => (),
+				}
+			}
+		});
+
+		SpawnedSubsystem {
+			name: "ForwardSubsystem",
```

Comment:
we use kebab-case elsewhere
```suggestion
			name: "forward-subsystem",
```

Reply:
Why? Dummy subsystem uses the same naming scheme?

Reply:
`rg` tells me that every other subsystem is in kebab case though so `DummySubsystem` may have slipped through on review or something idk. So maybe we should unify everything to one scheme, but i don't have a strong opinion on this one.

Reply:
yeah, `DummySubsystem` was an oversight

Reply:
I will change both 👍